web/modules/custom/dbcdk_community/src/Form/ProfileEditForm.php

```diff
@@ -0,0 +1,269 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\dbcdk_community\Form\ProfileEditForm.
+ */
+
+namespace Drupal\dbcdk_community\Form;
+
+use Drupal\dbcdk_community\CommunityTraits;
+use DBCDK\CommunityServices\Api\ProfileApi;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+
+/**
+ * Edit a Community Service Profile.
+ */
+class ProfileEditForm extends FormBase implements ContainerInjectionInterface {
+
+  use CommunityTraits;
+
+  /**
+   * The current request stack.
+   *
+   * @var RequestStack $requestStack
+   */
+  protected $requestStack;
+
+  /**
+   * The DBCDK Community Service Profile API.
+   *
+   * @var ProfileApi $profileApi
+   */
+  protected $profileApi;
+
+  /**
+   * The Community Service Profile object from the username context.
+   *
+   * @var Profile $profile
+   */
+  protected $profile;
+
+  /**
+   * The date format the Community Client expects.
+   *
+   * @var string $dateFormat
+   */
+  protected $dateFormat = 'Y-m-d';
+
+  /**
+   * Creates a Profile Edit Form instance.
+   *
+   * @param RequestStack $request_stack
+   *   The current request stack.
+   * @param ProfileApi $profile_api
+   *   The DBCDK Community Service Profile API.
+   */
+  public function __construct(RequestStack $request_stack, ProfileApi $profile_api) {
+    $this->profileApi = $profile_api;
+    $this->requestStack = $request_stack;
+    $this->profile = $this->getProfile($request_stack->getCurrentRequest()->get('username'));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('request_stack'),
+      $container->get('dbcdk_community.api.profile')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'dbcdk_community_profile_edit_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, $username = NULL) {
+    // Defensive coding to make sure the editor of the profile does not submit
+    // a form without default values (this could cause deletion of data if the
+    // editor was unaware of the lack of values).
+    if (empty($this->profile)) {
+      drupal_set_message($this->t('The system could not find any information about the profile at this moment. Please try and refresh or contact an administrator.'), 'error');
```

Comment:
The `handleException` method has been removed as stated in one of the comments above. But yes, that was previously the case - Amazing eyes! :eyes: